sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -189,6 +189,75 @@ async fn test_upload_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
+#[recorded::test]
+async fn test_delete_blob(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+
+    let recording = ctx.recording();
+    let (options, endpoint) = recorded_test_setup(recording);
+    let container_name = recording
+        .random_string::<17>(Some("container"))
+        .to_ascii_lowercase();
+    let blob_name = recording
+        .random_string::<12>(Some("blob"))
+        .to_ascii_lowercase();
+
+    let container_client_options = BlobContainerClientOptions {
+        client_options: options.clone(),
+        ..Default::default()
+    };
+    // Act
+    let container_client = BlobContainerClient::new(
+        &endpoint,
+        container_name.clone(),
+        recording.credential(),
+        Some(container_client_options),
+    )?;
+    container_client.create_container(None).await?;
+
+    let blob_client_options = BlobClientOptions {
+        client_options: options.clone(),
+        ..Default::default()
+    };
+    let blob_client = BlobClient::new(
+        &endpoint,
+        container_name,
+        blob_name,
+        recording.credential(),
+        Some(blob_client_options),
+    )?;
+
+    let data = b"hello rusty world";
+
+    blob_client
+        .upload(
+            RequestContent::from(data.to_vec()),
+            false,
+            u64::try_from(data.len())?,
+            None,
+        )
+        .await?;
+
+    // Assert
+    let response = blob_client.download(None).await?;
+    let content_length = response.content_length()?;
+    let (status_code, _, response_body) = response.deconstruct();
+    assert!(status_code.is_success());
+    assert_eq!(17, content_length.unwrap());
+    assert_eq!(Bytes::from_static(data), response_body.collect().await?);
```

Comment:
Seems like a bit overkill to download the Blob just to prove it's there. Maybe just get properties as that is typically used for existence checks.